[mqtt] Support Ruuvitags via Ruuvi Gateway (#13315)
openhab/openhab-addons: bundles/org.openhab.binding.mqtt.ruuvigateway/README.md

| movementCounter           | Number:Dimensionless     | Movement counter                                                         |
| rssi                      | Number                   | Received signal (between the Gateway and the sensor) strength indicator  |
| ts                        | DateTime                 | Timestamp when the message from Bluetooth-sensor was received by Gateway |
| gwts                      | DateTime                 | Timestamp when the message from Bluetooth-sensor was relayed by Gateway  |
| gwmac                     | String                   | MAC-address of Ruuvi Gateway                                             |

Note: not all channels are always updated.
Available fields depend on [Ruuvi Data Format](https://github.com/ruuvi/ruuvi-sensor-protocols).
At the time of writing (2022-09), most Ruuvi Tags use Ruuvi Data Format 5 out of box.

Some measurements might not make any sense.
For example, Ruuvi Tag Pro 2in1 does not have a humidity measurement and thus, the humidity data advertised by the sensor is garbage.

## Example

> Please note that Thing and Item configuration can be done fully in MainUI. 

Reviewer: @lolodomo - I remember seeing a comment somewhere to not explicitly mention MainUI, can you advise? Do we usually just refer to the "UI"?

Reviewer: OK. Just before you commented I committed a change for "MainUI" -> "the UI" to not have this small comment block the PR from being merged. That should be okay as well. I saw only very few references to "MainUI" when I grep'ed the repo yesterday and many *.md references to "UI".

> Please note that Thing and Item configuration can be done fully in MainUI. 

Reviewer: ```suggestion
Please note that Thing and Item configuration can be done fully in the UI.
```

## Setup the Gateway

Before using this binding, Ruuvi Gateway needs to configured to publish the sensor data via MQTT. 

For further instructions, refer to relevant section in [Ruuvi Gateway documentation](https://ruuvi.com/gateway-config/).
For most convenient usage of this binding, please ensure that "Use 'ruuvi' on the prefix' MQTT setting is enabled on Ruuvi Gateway.

## Discovery

First install the MQTT binding and setup a `broker` thing and make sure it is ONLINE, as this binding uses the MQTT binding to talk to your broker and hence that binding must be setup first.

This binding discovers the Ruuvi Tags via the MQTT bridge; the discovered things should appear in your thing Inbox.

## Thing Configuration

There is only thing type supported by this binding, `ruuvitag_beacon`.
No manual configuration is needed, and discovery function can be used instead.

For users that prefer manual configuration, we list here the configurable parameters.

| Parameter | Description                               | Required | Default |
|-----------|-------------------------------------------|----------|---------|
| `topic`   | MQTT topic containing the gateway payload | Y        | (N/A)   |

> 
> 

Reviewer: Excess lines.
```suggestion
```

| accelerationz             | Number:Acceleration      | The measured acceleration of Z                                           |
| txPower                   | Number:Power             | TX power                                                                 |
| dataFormat                | Number                   | Data format version                                                      |
| measurementSequenceNumber | Number:Dimensionless     | Measurement sequence number                                              |
| movementCounter           | Number:Dimensionless     | Movement counter                                                         |
| rssi                      | Number                   | Received signal (between the Gateway and the sensor) strength indicator  |
| ts                        | DateTime                 | Timestamp when the message from Bluetooth-sensor was received by Gateway |
| gwts                      | DateTime                 | Timestamp when the message from Bluetooth-sensor was relayed by Gateway  |
| gwmac                     | String                   | MAC-address of Ruuvi Gateway                                             |

> Note: not all channels are always updated. Available fields depend on [Ruuvi Data Format](https://github.com/ruuvi/ruuvi-sensor-protocols).

Reviewer: One sentence per line would make it easier to read diffs.
```suggestion
Note: not all channels are always updated.
Available fields depend on [Ruuvi Data Format](https://github.com/ruuvi/ruuvi-sensor-protocols).
```

At the time of writing (2022-09), most Ruuvi Tags use Ruuvi Data Format 5 out of box.

Some measurements might not make any sense.
For example, Ruuvi Tag Pro 2in1 does not have a humidity measurement and thus, the humidity data advertised by the sensor is garbage.

## Example

Please note that Thing and Item configuration can be done fully in MainUI. 
For those who prefer textual configuration, we share this example here.

To use these examples for textual configuration, you must already have a configured a MQTT `broker` thing and know its unique ID.
This UID will be used in the things file and will replace the text `myBroker`.
The first line in the things file will create a `broker` thing and this can be removed if you have already setup a broker in another file or via the UI already.

> *.things

Reviewer: ```suggestion
### *.things
```

> ```

Reviewer: ```suggestion
```java
```

Bridge mqtt:broker:myBroker [ host="localhost", secure=false, password="*******", qos=1, username="user"]
mqtt:ruuvitag_beacon:myTag1  "RuuviTag Sensor Beacon 9ABC" (mqtt:broker:myBroker) [ topic="ruuvi/mygw/DE:AD:BE:EF:AA:01" ]

> 

Reviewer: ```suggestion
```

```

> *.items

Reviewer: ```suggestion
### *.items
```

> ```

Reviewer: ```suggestion
```java
```
